### PR TITLE
[CI/Makefile] Parallelize windows unit tests

### PR DESCRIPTION
**Pull request by djaglowski**
Follows from #10020 and #10027

### CODE REVIEW

**Comment by dmitryax** [comment]
Nice! I was about to do the same

File: .github/workflows/build-and-test-windows.yml [hunk 1]
```diff
@@ -10,6 +10,39 @@ env:
...
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~\go\pkg\mod
+            ~\AppData\Local\go-build
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+      - name: Run Unit tests
+        run: make gotest GROUP=${{ matrix.group }}
+
+  windows-unittest (receiver-0):
```
Comment (dmitryax) [inline .github/workflows/build-and-test-windows.yml:45]: Is this left for debug purposes?

**Review by dmitryax — APPROVED** [review]

File: .github/workflows/build-and-test-windows.yml (hunk 1 above)
Comment (djaglowski) [reply .github/workflows/build-and-test-windows.yml]: It was an attempt to have this job match the name style of the others. It appears not to have worked though, and I think I've found a better way anyways.

**Comment by djaglowski** [comment]
The conditional steps appears to be working now:

image

image

**Pull request merged**